src/Mono.Android/Android.Runtime/InputStreamInvoker.cs

```diff
@@ -6,13 +6,18 @@ namespace Android.Runtime
 	public class InputStreamInvoker : Stream
 	{
 		public Java.IO.InputStream BaseInputStream {get; private set;}
+		public Java.Nio.Channels.FileChannel BaseFileChannel {get; private set;}
```

Comment:
*Should* this be public?  I'm not sure I see immediate benefit to making this a public API.

Reply:
You're right, will change to protected.

Reply:
Not sure I see much benefit to `protected` either.  All it does is prevent an `BaesInputStream as FileChannel` expression within the subclass, which doesn't feel like much benefit to me...

Reply:
I think it should be direct accessible from a subclass. For example in my project I was able to fix this issue by subclassing InputStreamInvoker and having access to the BaseInputStream. While it is true that even if it is marked as private, any subclass could just access it via ``Java.IO.FileInputStream fileStream = stream as Java.IO.FileInputStream; if (fileStream is object)``, I don't see the benefit in private here.
I don't understand your point about "All it does is prevent an BaesInputStream as FileChannel expression".
In the end, please let me know how you want it, it's your code base and I wanna finish this.